docs/subnets/create-a-vm-blobvm.md

```diff
@@ -0,0 +1,470 @@
+
+- **Non-Static Handlers** - They help in interacting with blockchains instantiated by the VM. The API's endpoint will be different for different chains. `/ext/bc/[chainID]`
+- **Static Handlers** - They help in directly accessing VM. These are optional for reasons such as parsing genesis bytes required to instantiate new blockchains. `/ext/vm/[vmID]`
+
+### Registering a VM
+
+While registering a VM, we store its **factory** against the user-defined **vmID**. As the name suggests, a VM factory can create new VM instances. Using the VM's instance, we can initialize it and can create a new chain. Note that, instantiating a VM is different than initializing it. We initialize the instance of a VM with parameters like genesis, database manager, etc. to create a functional chain.
+
+```go
+// Registering a factory inside avalanchego
+func (m *manager) RegisterFactory(vmID ids.ID, factory Factory) error {
+    ...
+    m.factories[vmID] = factory
+    ...
+}
+```
+
+### Instantiating a VM
+
+Each VM has a **factory** that is capable of creating new VM instances. A VM can be initialized as a chain along with handlers for accessing it, only when we have the VM's instance. The factory's `New` method shown below, returns the VM's instance to its caller in `avalanchego`.
+
+```go
+// Returning a new VM instance from VM's factory
+func (f *Factory) New(*snow.Context) (interface{}, error) { return &vm.VM{}, nil }
+```
+
+VM's methods and other parts of the code are made available to `avalanchego` by **registering** its factory and instantiating a new VM using this factory whenever a chain is required to be built. It's generally in the [`factory.go`](https://github.com/ava-labs/blobvm/blob/master/factory.go) file of the VM.
+
+```go
+// Instantiating a new VM from its factory in avalanchego
+func (m *manager) buildChain(chainParams ChainParameters, sb Subnet) (*chain, error) {
+    ...
+    vmFactory, err := m.VMManager.GetFactory(vmID)
+    vm, err := vmFactory.New(ctx.Context)
+    ...
+}
+```
+
+### Initializing a VM
+
+Chains are functional, only when the instantiated VMs are initialized. Initializing a VM involves setting up the database, block builder, mempool, genesis state, handlers, etc. This will expose the VM's API handlers, start accepting transactions, gossiping them across the network, building blocks, etc.
+
+```go
+if err := vm.Initialize(
+    ctx.Context,
+    vmDBManager,
+    genesisData,
+    chainConfig.Upgrade,
+    chainConfig.Config,
+    msgChan,
+    fxs,
+    sender,
+);
+```
+
+You can refer to the implementation of `vm.initialize` in the [BlobVM](https://github.com/ava-labs/blobvm/blob/master/vm/vm.go#L92) repository.
+
+### Interaction
+
+Once a VM is initialized, it becomes a functional chain. We can use the handlers to issue transactions, query chain state, and whatever functionalities are being provided by the VM.
```

Comment:
This phrase once a VM is initialized it becomes a functional chain is repeated a few places, but it seems pretty vague to me. Additionally, the VM is initialized before being registered to the chain manager and isn't that functional until bootstrapping is finished, so I think it would be better to be more specific here.